code/game/objects/items.dm

```diff
@@ -207,18 +207,16 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 
 /obj/item/ComponentInitialize()// this proc says it's for initializing components, but we're initializing elements too because it's you and me against the world >:)
 	. = ..()
-
-	if(embedding)
-		AddElement(/datum/element/embed, embedding)
-	else if(GLOB.embedpocalypse)
+	// this proc says it's for initializing components, but we're initializing elements too because it's you and me against the world >:)
+	if(!LAZYLEN(embedding) && GLOB.embedpocalypse)
 		embedding = EMBED_POINTY
-		AddElement(/datum/element/embed, embedding)
 		name = "pointy [name]"
-	else if(GLOB.stickpocalypse)
+	else if(!LAZYLEN(embedding) && GLOB.stickpocalypse)

```

Comment:
You can nest these checks under a `if(!LAZYLEN(embedding)` and avoid the duplicate check.